src/runtime/refiner.ts

```diff
@@ -606,13 +646,13 @@ export class Segment {
     let right: Boundary;
     if (a.from.val === b.from.val) {
       left = {...a.from};
-      left.kind = a.from.kind === b.from.kind ? a.from.kind : 'closed';
+      left.isOpen = a.from.isOpen === b.from.isOpen ? a.from.isOpen : false;
```

Comment:
This seems to be the same as `a.from.isOpen && b.to.isOpen`. Can you double check for me?